### PR TITLE
certificate-authority.md: create deployment doc based on yaml config.

### PR DESCRIPTION
**Pull request by Trishia**
@ondrejtomcik 
I newly created this guide based on yaml config as well as making certificates, getting/editing config file, running image without source download & build. Review and then give me comments. If good, merge plz.

### CODE REVIEW

**Comment by Trishia** [comment]
@ondrejtomcik 
Just moment plz. I will update minor fix.

**Comment by Trishia** [comment]
@ondrejtomcik 
I created this certificate-authority.md in docs/guide/deployment. 
Is it right place instead of certificate-authority/README.md or not ? If right, review all of them plz.

File: docs/guide/deployment/certificate-authority.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	-e CLOUD_SID=00000000-0000-0000-0000-000000000001 \
+	--entrypoint /bin/bash \
+	plgd/bundle:latest   
+
+# Copy & paste below commands on the bash shell of plgd/bundle container.
+certificate-generator --cmd.generateRootCA --outCert=/certs/root_ca.crt --outKey=/certs/root_ca.key --cert.subject.cn=RootCA 
+certificate-generator --cmd.generateCertificate --outCert=/certs/http.crt --outKey=/certs/http.key --cert.subject.cn=localhost --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
+certificate-generator --cmd.generateIdentityCertificate=$CLOUD_SID --outCert=/certs/coap.crt --outKey=/certs/coap.key --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
+cat /certs/http.crt > /certs/mongo.key
+cat /certs/http.key >> /certs/mongo.key
```
Comment (jkralik) [inline docs/guide/deployment/certificate-authority.md]: Pls remove them. They are not needed for certificate authority.

File: docs/guide/deployment/certificate-authority.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Certificate authority
+
```
Comment (jkralik) [inline docs/guide/deployment/certificate-authority.md]: Pls describe purpose of service.
This service is used to sign certificates for devices and plgd-dev/sdk (https://github.com/plgd-dev/sdk) client. 

BTW: During onboarding device via SDK (at first SDK gets OCF identity certificate for self), SDK ask the service to sign CSR (certificate signed request) for the device and when it is success the service returns certificate chain. 
eg plgd mobile application uses SDK for onboard the devices (https://play.google.com/store/apps/details?id=dev.plgd.client&hl=sk&gl=US) (https://apps.apple.com/sk/app/plgd/id1536315811)

File: docs/guide/deployment/certificate-authority.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+cat /certs/http.crt > /certs/mongo.key
+cat /certs/http.key >> /certs/mongo.key
+
+# Exit shell.
+exit 
+```
+```bash
+# See common certificates for plgd cloud services.
+ls .tmp/certs
+coap.crt	coap.key	http.crt	http.key	mongo.key	root_ca.crt	root_ca.key
```
Comment (jkralik) [inline docs/guide/deployment/certificate-authority.md]: Remove certs which are not generated after you remove lines (previos comment).

**Comment by jkralik** [comment]
> 
> 
> @ondrejtomcik
> I created this certificate-authority.md in docs/guide/deployment.
> Is it right place instead of certificate-authority/README.md or not ? If right, review all of them plz.

Yes. Could you create symlink and commit it?  
```bash
cd  certificate-authority
ln -sr ../docs/guide/deployment/certificate-authority.md README.md
```

File: docs/guide/deployment/certificate-authority.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
+# Certificate authority
+This service is used to sign certificates for devices and [plgd-dev/sdk](https://github.com/plgd-dev/sdk) client.
```
Comment (ondrejtomcik) [inline docs/guide/deployment/certificate-authority.md]: 'sign identity certificates'

File: docs/guide/deployment/certificate-authority.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+> During onboarding device via SDK (at first SDK gets OCF identity certificate for self), SDK ask the service to sign CSR (Certificate Signed Request) for the device and when it is success, the service returns certificate chain.
```
Comment (ondrejtomcik) [inline docs/guide/deployment/certificate-authority.md]: Initialization of the SDK requires an Identity Certificate, issued by this service after successful user authorization. The SDK is then able to discover and configure the ownership as described [here](https://openconnectivity.org/specs/OCF_Security_Specification_v2.2.1.pdf#page=38). During this process, a CSR (Certificate Signing Request) is sent to the Certificate Authority which returns device's Identity Certificate.

e.g. plgd mobile application uses SDK for device ownership configuration. Available on both [Google Play](https://play.google.com/store/apps/details?id=dev.plgd.client&hl=sk&gl=US) and [Apple Store](https://apps.apple.com/sk/app/plgd/id1536315811).

File: docs/guide/deployment/certificate-authority.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
+# Certificate authority
```
Comment (ondrejtomcik) [inline docs/guide/deployment/certificate-authority.md]: Certificate Authority

**Review by ondrejtomcik — APPROVED** [review]

**Pull request merged**